src/server/game/AI/SmartScripts/SmartScript.cpp

```diff
@@ -606,6 +606,10 @@ void SmartScript::ProcessAction(SmartScriptHolder& e, Unit* unit, uint32 var0, u
                         me->InterruptNonMeleeSpells(false);
                     }
 
+                    if (e.action.cast.castFlags & SMARTCAST_THREATLIST_NOT_SINGLE)
+                        if (me->GetThreatMgr().GetThreatListSize() == 1)
```

Comment:
There is probably never a case where threatlist can be 0? But just in case perhaps you should check <= 1?

Reply:
makes sense to me, thanks. I didn't like the ==